extensions/components/com_redcore/site/models/translation.php

```diff
@@ -0,0 +1,315 @@
+<?php
```

Comment:
instead of repeating the functions you should load admin model and reuse that code... you can load it with $model = RModel::getAdminInstance('translation', array(), 'com_redcore');